lib/live_cluster/show_controller.py

```diff
@@ -1878,3 +1881,63 @@ async def _do_default(self, line):
             self.cluster,
             **self.mods,
         )
+
+
+@CommandHelp(
+    "Displays client user agent information including version and application ID",
+    modifiers=(with_modifier_help,),
+    usage=f"[{ModifierUsage.WITH}]",
+)
+class ShowUserAgentsController(LiveClusterCommandController):
+    def __init__(self):
+        self.modifiers = set([Modifiers.WITH])
+        self.getter = GetUserAgentsController(self.cluster)
+
+    async def _do_default(self, line):
+        user_agents_data = await self.getter.get_user_agents(nodes=self.nodes)
+
+        # Process the user agents data
+        processed_data = {}
+        for node, agents in user_agents_data.items():
+            if isinstance(agents, Exception):
+                raise ShellException(
+                        f"Error processing user agent data from {node}: {agents}"
+                    )
+                
+            processed_data[node] = []
+            if not agents:
+                continue
+                
+            for agent in agents:
+                try:
+                    # Get user-agent and count from the agent dict
+                    user_agent = agent['user-agent']
+                    count = agent['count']
+                    
+                    # Decode base64 user agent
+                    decoded_ua = b64decode(user_agent).decode('utf-8')
+                    # Split by comma to get format-version, client-version, and app-id
+                    ua_parts = decoded_ua.split(',')
+                    
+                    if len(ua_parts) < 3:
+                        # If we don't have enough parts, mark as unknown
+                        version = "unknown"
+                        app_id = "unknown"
```

Comment:
What if a user sets their own user agent info and it doesn't match this format? Then asadm will display it as "unknown" right? Maybe asadm should be able to display user agents with formats different from its own? moreover what if multiple unrecognized user agent formats come back, won't each of them be displayed as "unknown"?

Reply:
All this may be fine if we always expect clients to follow this format.